ee/tabby-webserver/src/schema/email.rs

```diff
@@ -1,11 +1,120 @@
 use anyhow::Result;
 use async_trait::async_trait;
-use juniper::GraphQLObject;
+use juniper::{GraphQLEnum, GraphQLObject};
+use serde::Serialize;
+use validator::{validate_url, ValidationError};
+
+#[derive(GraphQLEnum, Clone, Debug)]
+pub enum Encryption {
+    StartTls,
+    SslTls,
+    None,
+}
+
+impl ToString for Encryption {
+    fn to_string(&self) -> String {
+        match self {
+            Encryption::StartTls => "STARTTLS",
+            Encryption::SslTls => "SSLTLS",
+            Encryption::None => "NONE",
+        }
+        .into()
+    }
+}
+
+#[derive(GraphQLEnum, Clone, Debug)]
+pub enum AuthMethod {
+    Plain,
+    Login,
+    XOAuth2,
+}
+
+impl ToString for AuthMethod {
+    fn to_string(&self) -> String {
+        match self {
+            AuthMethod::Plain => "PLAIN",
+            AuthMethod::Login => "LOGIN",
+            AuthMethod::XOAuth2 => "XOAUTH2",
+        }
+        .into()
+    }
+}
 
 #[derive(GraphQLObject)]
 pub struct EmailSetting {
     pub smtp_username: String,
     pub smtp_server: String,
+    pub from_address: String,
+    pub encryption: Encryption,
+    pub auth_method: AuthMethod,
+}
+
+fn validation_error<T: Serialize>(
+    msg: &'static str,
+    field: &'static str,
+    value: &T,
+) -> ValidationError {
+    let mut err = ValidationError::new(msg);
+    err.add_param(field.into(), value);
+    err
+}
+
+impl EmailSetting {
+    pub fn new_validate(
```

Comment:
I don't think we should do validate in this direction (the data is already in db, thus no need to implement it as a ValidationError).

I'll suggest:
1. keep the implementation, but return an `anyhow::Error`
2. In EmailService::get_email_setting, whenever a readed record is not valid (new_validate failed):
    - delete the email setting record from database.
    - return None instead.